src/contexts/Guilds/orbis.tsx

```diff
@@ -1,12 +1,78 @@
-import { createContext } from 'react';
 import { Orbis } from '@orbisclub/orbis-sdk';
+import { useState, useEffect, createContext } from 'react';
+import { useSigner } from 'wagmi';
+import { sleep } from 'utils';
 
 export const OrbisContext = createContext(null);
+const orbis = new Orbis();
 
 export const OrbisProvider = ({ children }) => {
-  const orbis = new Orbis();
+  const { data: signer } = useSigner();
+
+  const [profile, setProfile] = useState<any>(null);
+  const [hasLit, setHasLit] = useState(false);
+
+  const connectOrbis = async () => {
+    const res = await orbis.connect_v2({
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      provider: signer?.provider?.provider,
+    });
+
+    if (res.status !== 200) {
+      await sleep(2000);
+      await connectOrbis();
+    } else {
+      const { data } = await orbis.getProfile(res.did);
+      setProfile(data);
+    }
+  };
+
+  const disconnectOrbis = () => {
+    const res = orbis.logout();
+
+    if (res.status === 200) setProfile(null);
+  };
+
+  const connectLit = async () => {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment

```

Comment:
Why is this commented out?

Reply:
I tried using `provider` from wagmi's `useProvider` but it doesn't works on Orbis SDK. So instead I used `signer?.provider?.provider` which then throw ts error `Property 'provider' does not exist on type 'Provider'` but it worked as intended on frontend.

I'm not really sure why `provider` doesn't works, but that was the solution recommended by Orbis team.